File: Crio/crio_programming_interview_problems-master/TheMesseuse/TheMesseuse.java
import java.util.*;


class TheMesseuse{
	public static long theMesseuse(int [] time,int len)
	{
		// CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
		long [] dp= new long [len];
		dp[0]= time[0];
		dp[1] = time[1];
		for (int i = 2; i < len; i++) {
			dp[i] = Math.max(dp[i - 1], dp[i - 2] + time[i]);
		}
		return dp[len-1];
		// CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
	}

	public static void main(String[] args) {
		Scanner sc = new Scanner(System.in);
		int n = sc.nextInt();
		int[] ar = new int[n];
		for (int i = 0; i < n; i++)
			ar[i] = sc.nextInt();
		System.out.println(theMesseuse(ar, n));
	}
}

File: Crio/crio_programming_interview_problems-master/TheMesseuse/TheMesseuse.cpp
#include<bits//stdc++.h>
using namespace std;
long long theMesseuse( vector<long long> A) {
	// CRIO_SOLUTION_START_MODULE_L1_PROBLEMS
	int n = A.size();
	vector<long long> dp(n + 4);
	dp[0] = A[0];
	dp[1] = A[1];
	for ( int i = 2; i < n ; i++) {
		dp[i] = max( dp[i - 1] , dp[i - 2] + A[i]);
	}
	return  dp[n - 1];
	// CRIO_SOLUTION_END_MODULE_L1_PROBLEMS
}
int main() {

	int n;
	cin >> n;
	vector<long long> A(n);
	for ( int i = 0 ; i < n ; i++) {
		cin >> A[i];
	}
	cout << theMesseuse(A) << endl;


}